core/src/main/java/org/bitcoinj/core/ECKey.java

```diff
@@ -417,6 +418,19 @@ public boolean isCompressed() {
         return pub.isCompressed();
     }
 
+    public Address toAddress(ScriptType scriptType, BitcoinNetwork network) {
```

Comment:
Can also be in a future PR, just wanted to throw in the idea.

Reply:
> It's a bit unfortunate that you'll need to cast here in some cases.

You mean cast from `Network` to `BitcoinNetwork`? I suppose we could have this method accept `Network` and throw `IllegalArgumentException` (or maybe even `ClassCastException) if it's not a `BitcoinNetwork`. It's functionally equivalent, but less verbose for callers.

There may also be a more elegant long-term solution using an interface (something similar to what we are doing in PR #2601 with `AddressParser`)  -- Note that in a previous PR I had an `AddressKey` interface that defined this `toAddress` method and we could do something like that in the near future if it looks good.

Reply:
> I wonder if we could offer `public T toAddress(Class<T extends Address>, BitcoinNetwork network)`, either in addition to what this PR already has or instead of.

Yes, let's consider this in the near future.

Reply:
I was thinking of a statement like `(SegwitAddress) key.toAddress(ScriptType.P2WPKH, params.network())`.

You have to specify the script type, and then (somewhat redundantly) cast the result to `SegwitAddress`. Of course, normally you should just use `Address` to store (makes your code more future proof), but still it's a bit of a nuisance.